lib/stringify.js

```diff
@@ -65,9 +65,9 @@ internals.stringify = function (object, prefix, generateArrayPrefix, strictNullH
         }
 
         if (Array.isArray(obj)) {
-            values = values.concat(internals.stringify(obj[key], generateArrayPrefix(prefix, key), generateArrayPrefix, strictNullHandling, skipNulls, encode, filter));
+            values = values.concat(internals.stringify(obj[key], generateArrayPrefix(prefix, key), generateArrayPrefix, strictNullHandling, skipNulls, encode, filter, sort, allowDots));
```

Comment:
it looks like you caught a minor bug/omission with the missing "sort" param here and below - i'm not sure if that was intentionally omitted or not originally. could you also add tests that cover the presence/absence of this parameter (here and below)?

Alternatively, if you could prepare a separate PR that adds the sort param and the tests for it, i'd prefer to merge that first and keep this PR focused on "allowDots" :-)